src/OrderFormAcceptanceTests.Steps/Steps/PreviewOrderSummary.cs

```diff
@@ -1,344 +1,393 @@
+            orderItem.EstimationPeriodId = 1;
+            orderItem.TimeUnitId = 1;
+            orderItem.Create(Test.ConnectionString);
+            Context.Add("CreatedOrderItem", orderItem);
+        }
+
+        [Given(@"an additional service with a flat price variable \(Patient\) order type is saved to the order")]
+        public void GivenAnAdditionalServiceWithAFlatPriceVariablePatientOrderTypeIsSavedToTheOrder()
+        {
+            SetOrderCatalogueSectionToComplete();
+            new AdditionalServices(Test, Context).GivenTheSupplierAddedToTheOrderHasAnAdditionalServiceWithAPatientFlatPrice();
+            var orderItem = new OrderItem().GenerateAdditionalServiceOrderItemWithVariablePricedPerPatient((Order)Context["CreatedOrder"]);
+            orderItem.Create(Test.ConnectionString);
+            Context.Add("CreatedOrderItem", orderItem);
+        }
+
+        [Given(@"there are one or more Order items summarised in the Order items \(recurring cost\) table")]
+        public void GivenThereAreOneOrMoreOrderItemsSummarisedInTheOrderItemsRecurringCostTable()
+        {
+            var onDemandOrderItem = new OrderItem().GenerateOrderItemWithFlatPricedVariableOnDemand((Order)Context["CreatedOrder"]);
+            onDemandOrderItem.Create(Test.ConnectionString);
+
+            var declarativeOrderItem = new OrderItem().GenerateOrderItemWithFlatPricedVariableDeclarative((Order)Context["CreatedOrder"]);
+            declarativeOrderItem.Create(Test.ConnectionString);
+
+            var patientOrderItem = new OrderItem().GenerateOrderItemWithFlatPricedVariablePerPatient((Order)Context["CreatedOrder"]);
+            patientOrderItem.Create(Test.ConnectionString);
+        }
+
+        [Then(@"the Total cost for one year is the result of the Total cost for one year calculation")]
+        public void ThenTheTotalCostForOneYearIsTheResultOfTheTotalCostForOneYearCalculation()
+        {
+            var actual = Test.Pages.OrderForm.GetTotalAnnualCost();
+
+            const string expectedTotalAnnualCost = "1,316,535.00";
+            actual.Should().Be(expectedTotalAnnualCost);
+        }
+
+        [Then(@"the Total cost for one year is expressed as two decimal places")]
+        public void ThenItIsExpressedAsTwoDecimalPlaces()
+        {
+            const int decimalPointPartIndex = 1;
+            const int expectedDecimalPointLength = 2;
+
+            var actual = Test.Pages.OrderForm.GetTotalAnnualCost().Split('.')[decimalPointPartIndex].Length;
+
+            actual.Should().Be(expectedDecimalPointLength);
+        }
+
+        [Then(@"the Total monthly cost is the result of the Total monthly cost calculation")]
+        public void ThenTheTotalMonthlyCostIsTheResultOfTheTotalMonthlyCostCalculation()
+        {
+            var actual = Test.Pages.OrderForm.GetTotalMonthlyCost();
+
+            const string expectedTotalAnnualCost = "109,711.25";

```

Comment:
not sure about having this hard coded, is there anyway we can calc it based on what we know?  like ThenTheItemYearCostOfEachItemIsTheResultOfTheFlatCalculationPriceQuantityTimes12RoundedUpToTwoDecimalPlaces

Reply:
Yeah I was torn about the hardcoded expected value. Initially Jon & I read all the order items in the recurring cost table & added them all together. Then used this value as the expected but because of the rounding we had to put an approximate check which put me off slightly. I can go back to this though.

The step `ThenTheItemYearCostOfEachItemIsTheResultOfTheFlatCalculationPriceQuantityTimes12RoundedUpToTwoDecimalPlaces` is a useful reference the problem as it stands the context it is using a single order item `(OrderItem)Context["CreatedOrderItem"];`. I have a list of order items. So wouldn't be able to reuse the step as it stands. Also having to duplicate the custom cost calculation per order item was a little daunting, especially with the different combinations of provisioning type / catalogue item type etc.. but I suppose you already have it in places.

Open to suggestions...

Reply:
I personally am not a fan, however i can see the justification if it's going to take days to calculate it in code. @PhysicsJon what's your view?
If the data changes how long will it take us to manually recalculate it and update this value?

Reply:
I would much prefer a calculated cost rather than hard coded. Hard coded just screams fragility to me